webapp/src/js/components/Panoptes.js

```diff
@@ -58,71 +65,86 @@ let Panoptes = React.createClass({
     };
   },
 
+  handleResize() {
+    this.getFlux().actions.session.appResize();
+  },
+
   render() {
     let actions = this.getFlux().actions.session;
     let {tabs, popups, modal, components} = this.state.session.toObject();
     modal = modal.toObject();
     let userID = this.state.panoptes.getIn(['user', 'id']);
     let config = this.getConfig();
     return (
-      <MuiThemeProvider muiTheme={muiTheme}>
-        <div>
-          <div className="loading-container">
-            <div className="spinner" />
-          </div>
-          <div className="page">
-            <Header name={config.settings.name} userID={userID} logo={config.logo}/>
-            <div className="body">
-              <TabbedArea activeTab={tabs.get('selectedTab')}
-                          unclosableTab={tabs.get('unclosableTab')}
-                          onSwitch={actions.tabSwitch}
-                          onClose={actions.tabClose}
-                          onAddTab={actions.tabOpen}
-                          onDragAway={actions.tabPopOut}
-              >
-                {tabs.get('components').map((compId) => {
-                  let tab = components.get(compId).toObject();
-                  let props = tab.props ? tab.props.toObject() : {};
-                  props.componentUpdate = actions.componentUpdateFor(compId);
-                  return (
-                    <TabPane
-                      compId={compId}
-                      key={compId}>
-                      {React.createElement(dynamicRequire(tab.component), props)}
-                    </TabPane>
-                  );
-                })}
-              </TabbedArea>
+      <DetectResize onResize={this.handleResize}>
```

Comment:
We could use window.onresize here. Only works if <Panoptes> is whole page though I guess. MIght be an OK assumption.